modules/business-text/business-text.service.js

```diff
@@ -0,0 +1,64 @@
+const BusinessText = require('./business-text.model');
+const {
+  BUSINESS_TEXT_NOT_FOUND,
+  BUSINESS_TEXT_ALREADY_EXIST,
+} = require('../../error-messages/business-text.messages');
+
+class BusinessTextService {
+  async getAllBusinessTexts() {
+    return await BusinessText.find();
+  }
+
+  async getBusinessTextById(id) {
+    const businessText = await BusinessText.findById(id);
+    if (businessText) {
+      return businessText;
+    }
+    throw new Error(BUSINESS_TEXT_NOT_FOUND);
+  }
+
+  async getBusinessTextByCode(code) {
+    const businessText = await BusinessText.findOne({ code });
+    if (businessText) {
+      return businessText;
+    }
+    throw new Error(BUSINESS_TEXT_NOT_FOUND);
+  }
+
+  async updateBusinessText(id, businessText) {
+    if (await this.checkBusinessTextExist(businessText, id)) {
```

Comment:
Please, add a check that business text exists by id before performing any other checks